Library/ViewModels/LiveStreamChatViewModel.swift

```diff
@@ -0,0 +1,223 @@
+          let userId = liveStreamEvent.firebase?.chatUserId,
+          let name = liveStreamEvent.firebase?.chatUserName,
+          let avatar = liveStreamEvent.firebase?.chatAvatarUrl,
+          let token = liveStreamEvent.firebase?.token
+          else { return nil }
+
+        return LiveStreamChatUserInfo(name: name,
+                                      profilePictureUrl: avatar,
+                                      userId: userId,
+                                      token: token)
+      }
+      .skipNil()
+
+    self.didAuthorizeChat = liveAuthTokenFetch.filter { $0.isTerminating }.map { $0.error == nil }
+    self.updateLiveAuthTokenInEnvironment = liveAuthTokenFetch.values().map { $0.token }
+    self.willAuthorizeChat = shouldFetchAuthToken
+
+    self.chatInputViewHidden = liveStreamEvent.map { $0.liveNow }.map(negate)
+    self.dismissKeyboard = self.deviceOrientationDidChangeProperty.signal.ignoreValues()
+  }
+
+  private let chatInputViewRequestedLoginProperty = MutableProperty()
+  public func chatInputViewRequestedLogin() {
+    self.chatInputViewRequestedLoginProperty.value = ()
+  }
+
+  private let configData = MutableProperty<(Project, LiveStreamEvent, Bool)?>(nil)
+  public func configureWith(project: Project, liveStreamEvent: LiveStreamEvent, chatHidden: Bool) {
+    self.configData.value = (project, liveStreamEvent, chatHidden)
+  }
+
+  private let deviceOrientationDidChangeProperty = MutableProperty<UIInterfaceOrientation?>(nil)
+  public func deviceOrientationDidChange(orientation: UIInterfaceOrientation) {
+    self.deviceOrientationDidChangeProperty.value = orientation
+  }
+
+  private let didSetChatHiddenProperty = MutableProperty(false)
+  public func didSetChatHidden(hidden: Bool) {
+    self.didSetChatHiddenProperty.value = hidden
+  }
+
+  private let moreMenuButtonTappedProperty = MutableProperty()
+  public func moreMenuButtonTapped() {
+    self.moreMenuButtonTappedProperty.value = ()
+  }
+
+  private let viewDidLoadProperty = MutableProperty()
+  public func viewDidLoad() {
+    self.viewDidLoadProperty.value = ()
+  }
+
+  private let receivedChatMessagesProperty = MutableProperty<[LiveStreamChatMessage]?>(nil)
+  public func received(chatMessages: [LiveStreamChatMessage]) {
+    self.receivedChatMessagesProperty.value = chatMessages
+  }
+
+  private let userSessionStartedProperty = MutableProperty()
+  public func userSessionStarted() {
+    self.userSessionStartedProperty.value = ()
+  }
+
+  public let chatInputViewHidden: Signal<Bool, NoError>
+  public let configureChatHandlerWithUserInfo: Signal<LiveStreamChatUserInfo, NoError>
+  public let dismissKeyboard: Signal<(), NoError>
+  public let didAuthorizeChat: Signal<Bool, NoError>
+  public let openLoginToutViewController: Signal<LoginIntent, NoError>
+  public let presentMoreMenuViewController: Signal<(LiveStreamEvent, Bool), NoError>
+  public let prependChatMessagesToDataSource: Signal<([LiveStreamChatMessage], Bool), NoError>
```

Comment:
could either create a custom struct for this tuple to describe what the `Bool` is for, or could call the output `prependChatMessagesToDataSourceAndReload`